src/WebJobs.Extensions.DurableTask/Bindings/OrchestrationTriggerAttributeBindingProvider.cs

```diff
@@ -100,52 +100,66 @@ private class OrchestrationTriggerBinding : ITriggerBinding
+            public Task<ITriggerData> BindAsync(object? value, ValueBindingContext context)
             {
-                var orchestrationContext = (DurableOrchestrationContext)value;
-                Type destinationType = this.parameterInfo.ParameterType;
-
-                object convertedValue = null;
-                if (destinationType == typeof(IDurableOrchestrationContext))
+                if (value is DurableOrchestrationContext orchestrationContext)
                 {
-                    convertedValue = orchestrationContext;
+                    Type destinationType = this.parameterInfo.ParameterType;
+
+                    object? convertedValue = null;
+                    if (destinationType == typeof(IDurableOrchestrationContext))
+                    {
+                        convertedValue = orchestrationContext;
+                    }
+                    else if (this.config.TypedCodeProvider.IsInitialized &&
+                        destinationType.Name == TypedCodeProvider.ITypedDurableOrchestrationContext)
+                    {
+                        convertedValue = this.config.TypedCodeProvider.InstantiateTypedDurableOrchestrationContext(orchestrationContext);
+                    }
+                    else if (destinationType == typeof(string))
+                    {
+                        convertedValue = OrchestrationContextToString(orchestrationContext);
+                    }
+
+                    var inputValueProvider = new ObjectValueProvider(
+                        convertedValue ?? value,
+                        this.parameterInfo.ParameterType);
+
+                    var bindingData = new Dictionary<string, object?>(StringComparer.OrdinalIgnoreCase)
+                    {
+                        [this.parameterInfo.Name!] = convertedValue,
+                    };
+
+                    // We don't specify any return value binding because we process the return value
+                    // earlier in the pipeline via the InvokeHandler extensibility.
+                    var triggerData = new TriggerData(inputValueProvider, bindingData);
+                    return Task.FromResult<ITriggerData>(triggerData);
                 }
-                else if (this.config.TypedCodeProvider.IsInitialized &&
-                    destinationType.Name == TypedCodeProvider.ITypedDurableOrchestrationContext)
+                else if (value is RemoteOrchestratorContext remoteContext)
                 {
-                    convertedValue = this.config.TypedCodeProvider.InstantiateTypedDurableOrchestrationContext(orchestrationContext);
+                    // Remote context is only for modern out-of-process function execution and
+                    // contains a lighter payload.
+                    string serializedContext = JsonConvert.SerializeObject(remoteContext, JsonSettings);
+                    var valueProvider = new ObjectValueProvider(serializedContext, typeof(string));
```

Comment:
nit - the name `valueProvider` isn't terribly descriptive. Would `contextValueProvider` be more accurate?